runtime/bcverify/staticverify.c

```diff
@@ -1893,8 +1939,16 @@ j9bcv_verifyClassStructure (J9PortLibrary * portLib, J9CfrClassFile * classfile,
 		}
 		if (isInit) {
 			BOOLEAN invalidRetType = FALSE;
-			if ((CFR_METHOD_NAME_INIT == isInit) /* Check return type of <init> on VTs, it will eventually be changed to <new>. */
-				&& J9_IS_CLASSFILE_VALUETYPE(classfile)
+			if (
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+#ifdef J9VM_OPT_VALHALLA_NEW_FACTORY_METHOD
+				(CFR_METHOD_NAME_NEW == isInit)
+#else /* #ifdef J9VM_OPT_VALHALLA_VALUE_TYPES */
+				(CFR_METHOD_NAME_INIT == isInit) && J9_IS_CLASSFILE_VALUETYPE(classfile)
+#endif /* #ifdef J9VM_OPT_VALHALLA_VALUE_TYPES */
+#else /* #ifdef J9VM_OPT_VALHALLA_NEW_FACTORY_METHOD */
+				FALSE
```

Comment:
I thought the condition for the non-valhalla build would always evaluate to `FALSE`, since the code was `(CFR_METHOD_NAME_INIT == isInit) && J9_IS_CLASSFILE_VALUETYPE(classfile)` which should always expand to `(CFR_METHOD_NAME_INIT == isInit) && FALSE`

Reply:
OK.